sdks/python/apache_beam/runners/portability/portable_runner.py

```diff
@@ -445,15 +446,22 @@ def start(cls, worker_threads=1):
 
   def NotifyRunnerAvailable(self, start_worker_request, context):
     try:
-      worker = sdk_worker.SdkHarness(
-          start_worker_request.control_endpoint.url,
-          worker_count=self._worker_threads,
-          worker_id=start_worker_request.worker_id)
-      worker_thread = threading.Thread(
-          name='run_worker_%s' % start_worker_request.worker_id,
-          target=worker.run)
-      worker_thread.daemon = True
-      worker_thread.start()
+      command = ['python', '-c',
+                 'from apache_beam.runners.worker.sdk_worker '
+                 'import SdkHarness as SH; '
+                 'SH("%s",worker_count=%d,worker_id="%s").run()' % (
+                     start_worker_request.control_endpoint.url,
+                     self._worker_threads,
+                     start_worker_request.worker_id), ]
+      logging.warn("Starting worker with command %s" % (command))
+      process = subprocess.Popen(command, stdout=subprocess.PIPE, )
+
+      def kill_subprocess():
+        process.kill()
```

Comment:
Perhaps forcefully kill if it does not exit?

Reply:
There is no parameter for forceful kill. I suppose the kill method will ensure that the subprocress is killed.